File: Python/Motorcycle.py
# import class ParkingLot dari file ParkingLot
from ParkingLot import ParkingLot

# deklarasi kelas

class Motorcycle (ParkingLot):
    # atribut private
    __jenis_motor = ""
    __kap_tangki = ""

   # konstruktor
    def __init__(self, jenis_motor="", kap_tangki=""):
        super().__init__(jenis_motor, kap_tangki)
        self.__jenis_motor = jenis_motor
        self.__kap_tangki = kap_tangki

    # Getter & Setter
    
    def get_jenis_motor(self):
        return self.__jenis_motor

    def set_jenis_motor(self, jenis_motor):
        self.__jenis_motor = jenis_motor

    def get_kap_tangki(self):
        return self.__kap_tangki

    def set_kap_tangki(self, kap_tangki):
        self.__kap_tangki = kap_tangki

File: CPP/Motorcycle.cpp
// import file header
#include "header.hh"

/* konstruktor */

// konstruktor tanpa parameter
Motorcycle::Motorcycle()
{
    // set isi atribut secara default
    this->jenis_motor = "";
    this->kapasitas_tangki = "";
}
// konstruktor dengan parameter
Motorcycle::Motorcycle(string plat, string merk, string tahun_produksi, string warna, string jenis_motor, string kapasitas_tangki) : Garage(plat, merk, tahun_produksi, warna, merk_garasi, luas_garasi)
{
    // set isi atribut berdasarkan isi parameter
    this->jenis_motor = jenis_motor;
    this->kapasitas_tangki = kapasitas_tangki;
}

/* Getter dan Setter*/

// Get jenis_motor
string Motorcycle::get_jenis_motor()
{
    return this->jenis_motor;
}

// Set jenis_motor
void Motorcycle::set_jenis_motor(string jenis_motor)
{
    this->jenis_motor = jenis_motor;
}

// Get kapasitas_tangki
string Motorcycle::get_kapasitas_tangki()
{
    return this->kapasitas_tangki;
}

// Set kapasitas_tangki
void Motorcycle::set_kapasitas_tangki(string kapasitas_tangki)
{
    this->kapasitas_tangki = kapasitas_tangki;
}

/* desktruktor */

// default
Motorcycle::~Motorcycle() // desktruktor
{
}
